packages/bindings/src/mocks/mock_queriers.rs

```diff
@@ -1,51 +1,331 @@
+    ///             report: MockReportsQueries::get_mocked_report(subspace_id),
+    ///         })
+    ///         .into(),
+    ///         _ => ContractResult::Err("not supported".to_string()),
+    ///     });
+    /// ```
+    #[cfg(feature = "reports")]
+    pub fn with_custom_reports_handler<CH>(mut self, handler: CH) -> Self
+    where
+        CH: Fn(&ReportsQuery) -> ContractResult<Binary> + 'static,
+    {
+        self.reports_handler = MockDesmosQuerier::wrap_handler(handler);
+        self
+    }
+
+    /// Function to provide an user defined handler to mock responses to requests made to
+    /// the desmos `reactions` module.
+    /// * `handler` - Function that will be called when the contract under test performs a query
+    /// towards the `reactions` module.
+    ///
+    /// # Example
+    /// ```
+    /// use cosmwasm_std::{ContractResult, to_binary};
+    /// use desmos_bindings::mocks::mock_queriers::MockDesmosQuerier;
+    /// use desmos_bindings::reactions::mocks::MockReactionsQueries;
+    /// use desmos_bindings::reactions::models_query::QueryReactionResponse;
+    /// use desmos_bindings::reactions::query::ReactionsQuery;
+    ///
+    /// let querier =
+    ///     MockDesmosQuerier::default().with_custom_reactions_handler(|query| match query {
+    ///         ReactionsQuery::Reaction { subspace_id, post_id, reaction_id } => to_binary(&QueryReactionResponse {
+    ///             reaction: MockReactionsQueries::get_mock_reaction(),
+    ///         })
+    ///         .into(),
+    ///         _ => ContractResult::Err("not supported".to_string()),
+    ///     });
+    /// ```
+    #[cfg(feature = "reactions")]
+    pub fn with_custom_reactions_handler<CH>(mut self, handler: CH) -> Self
+    where
+        CH: Fn(&ReactionsQuery) -> ContractResult<Binary> + 'static,
+    {
+        self.reactions_handler = MockDesmosQuerier::wrap_handler(handler);
+        self
+    }
+}
+
+impl Querier for MockDesmosQuerier {
+    fn raw_query(&self, bin_request: &[u8]) -> QuerierResult {
+        let request: QueryRequest<DesmosQuery> = match from_slice(bin_request) {
+            Ok(v) => v,
+            Err(e) => {
+                return SystemResult::Err(SystemError::InvalidRequest {
+                    error: format!("Parsing query request: {}", e),
+                    request: bin_request.into(),
+                })
+            }
+        };
+        self.handle_query(&request)
+    }
+}
+
+impl Default for MockDesmosQuerier {
+    fn default() -> Self {
+        MockDesmosQuerier::new(&[])
+    }
+}
+
 /// Replacement for cosmwasm_std::testing::mock_dependencies
 /// this use our CustomQuerier to use desmos querier
 pub fn mock_dependencies_with_custom_querier(
     contract_balance: &[Coin],
-) -> OwnedDeps<MockStorage, MockApi, MockQuerier<DesmosQuery>, DesmosQuery> {
+) -> OwnedDeps<MockStorage, MockApi, MockDesmosQuerier, DesmosQuery> {
```

Comment:
I think we can provide `querier` argument to build custom querier deps quickly.
```rust
pub fn mock_dependencies_with_custom_querier(
    contract_balance: &[Coin], querier: MockDesmosQuerier,
) -> OwnedDeps<MockStorage, MockApi, MockDesmosQuerier, DesmosQuery> {
```

Then having another function to mock default deps, say:
```rust
pub type DesmosDeps =  OwnedDeps<MockStorage, MockApi, MockDesmosQuerier, DesmosQuery>;
pub fn mock_default_dependencies() ->  DesmosDeps 
```
What do you think?

Reply:
@dadamu Agree on the firts, for the second maybe is better `mock_desmos_dependencies` so the name is similar to the cosmwasm `mock_dependencies` function, sounds good to you?

Reply:
Yeah, it would be better to use `mock_desmos_dependencies` to avoid the name conflict.

Reply:
I have also renamed `mock_dependencies_with_custom_querier` to `mock_desmos_dependencies_with_custom_querier` to keep a consisten naming between functions, note that since we are using our custom querier in `mock_desmos_dependencies_with_custom_querier` the `contract_balance: &[Coin]` argument is not required anymore and so this PR creates a compatibility issue with the previous verions.